src/WebAdmin.cpp

```diff
@@ -489,86 +475,93 @@ AString cWebAdmin::GetContentTypeFromFileExt(const AString & a_FileExtension)
 
 sWebAdminPage cWebAdmin::GetPage(const HTTPRequest & a_Request)
 {
-	sWebAdminPage Page;
-	AStringVector Split = StringSplit(a_Request.Path, "/");
+	sWebAdminPage page;
+	auto split = StringSplit(a_Request.Path, "/");
 
-	// Find the plugin that corresponds to the requested path
-	AString FoundPlugin;
-	if (Split.size() > 1)
+	// If no specific page was requested, return an empty object:
+	if (split.size() <= 2)
 	{
-		for (PluginList::iterator itr = m_Plugins.begin(); itr != m_Plugins.end(); ++itr)
+		return page;
+	}
+
+	// Find the WebTab handler responsible for the request:
+	cWebTabPtr tab;
 		{
-			if ((*itr)->GetWebTitle() == Split[1])
+		cCSLock Lock(m_CS);
+		for (auto & wt: m_WebTabs)
+			{
+			if (
+				(wt->m_PluginName == split[1]) &&
+				(wt->m_UrlPath == split[2])
+			)
 			{
-				Page.Content = (*itr)->HandleWebRequest(a_Request);
-				cWebPlugin * WebPlugin = *itr;
-				FoundPlugin = WebPlugin->GetWebTitle();
-				AString TabName = WebPlugin->GetTabNameForRequest(a_Request).first;
-				Page.PluginName = FoundPlugin;
-				Page.TabName = TabName;
+				tab = wt;
 				break;
 			}
 		}
 	}
 
-	// Return the page contents
-	return Page;
+	// If a WebTab handler was found, call it:
+	if (tab != nullptr)
+	{
+		page.ContentType = "text/html";  // Default to HTML content type, unless overridden by a plugin
```

Comment:
Does this mean plugins can change the content type even for 'normal' pages - eg, not with `~`?